diff --git i/caddyhttp/httpserver/server.go w/caddyhttp/httpserver/server.go
index 800f921..8df2697 100644
--- i/caddyhttp/httpserver/server.go
+++ w/caddyhttp/httpserver/server.go
@@ -376,8 +376,6 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	c = context.WithValue(r.Context(), ReplacerCtxKey, replacer)
 	r = r.WithContext(c)
 
-	w.Header().Set("Server", caddy.AppName)
-
 	status, _ := s.serveHTTP(w, r)
 
 	// Fallback error response in case error handling wasn't chained in
